LocalPackages/DataBrokerProtection/Sources/DataBrokerProtection/Scheduler/DataBrokerProtectionAgentManager.swift

```diff
@@ -194,23 +222,23 @@ extension DataBrokerProtectionAgentManager: DataBrokerProtectionAgentAppEvents {
 
         userNotificationService.requestNotificationPermission()
         fireMonitoringPixels()
-        queueManager.startImmediateOperationsIfPermitted(showWebView: false, operationDependencies: operationDependencies) { [weak self] errors in
+        queueManager.startImmediateScanOperationsIfPermitted(showWebView: false, operationDependencies: operationDependencies) { [weak self] errors in
```

Comment:
We discussed over MM, but posting here for prosperity:
- Previously (i.e. prior to this PR), things higher up the stack (agent manager, queueManager) would tell stuff lower (e.g. DataBrokerOperation) to run "immediate operations" when a profile is saved. I.e. it was non specific about what immediate operations included
- Now, they explicitly tell it only to do scans (and are now specific for the kinds of scans)
- This is enforced through DataBrokerOperation.filterAndSortOperationsData()
- Ignoring freemium, this actually represents mostly no change in practice for subscribed users, because for manual scans we never actually ran opt outs immedtiely anyway (which I think might actually be an implementation mistake, but seems to be the observed behavior)
  
So now instead AgentManager is explicit about what immediate operations includes, which I think is an improvement and where the decision should be made.
  
  But the main reason I mention it is because I think that original behavior is a bug and we do want opt outs to run immediately after the scans,  so it's worth discussing if we want to codify it like this.
    
 I think we do, because of how I think we should fix that bug, which is on the completion for the manual scans, in agent manager, call the thing on queuemanager, rather than waiting for the scheduler to do it. But there's a slight asterisk of we'd need to move something around in how the QueueManager handles completions. atm it would break some stuff cos it calls the completion before clearing errors
      
  The alternative approach is to somehow add the opt outs to the queue as we make them. If it was easy to do I'd seriously consider doing it, but as it would be much much harder and be a big increase in implementation complexity lower down the stack solely for making AgentManager slightly clearer.

Reply:
Thanks for capturing this here @THISISDINOSAUR . You set a really good standard for prosperity, it’s great.